link-check.js

```diff
@@ -23,37 +33,48 @@ function allMatches(str, regex) {
 
 let invalid = 0;
 
-for (const file of files) {
-  const filePretty = chalk.bold(
-    path
-      .relative(__dirname, file)
-      .replace(/^public/, "")
-      .replace(/index.html$/, "")
-  );
-  const contents = fs.readFileSync(file, "utf8");
-  if (contents.indexOf("Postgraphile") >= 0) {
-    invalid++;
-    console.error(
-      `${filePretty} mentions 'Postgraphile'; please change to 'PostGraphile' or 'postgraphile' for consistency.`
+const fileLinks = files
+  .map(file => {
+    const filePretty = chalk.bold(
+      path
+        .relative(__dirname, file)
+        .replace(/^public/, "")
+        .replace(/index.html$/, "")
     );
-  }
-  const links = allMatches(contents, /<a[^>]+href="([^"]+)"/g);
-  for (const link of links) {
+    const contents = fs.readFileSync(file, "utf8");
+    if (contents.indexOf("Postgraphile") >= 0) {
+      invalid++;
+      console.error(
+        `${filePretty} mentions 'Postgraphile'; please change to 'PostGraphile' or 'postgraphile' for consistency.`
+      );
+    }
+
+    // WARNING! Cardinal sin below, shield eyes
+    return allMatches(contents, /<a[^>]+href="([^"]+)"/g).map(link => {
+      return {
+        link: entities.decode(link),
+        filePretty,
+      };
+    });
+  })
+  .reduce((flattened, toFlatten) => flattened.concat(toFlatten), []);
+
+pMap(
+  fileLinks,
+  async ({ filePretty, link }) => {
     const trimmed = link.replace(/[?#].*$/, "");
-    if (trimmed.match(/\.(js|css|png|svg|webmanifest)$/)) {
+
+    if (trimmed.match(/\.(css|png|svg|webmanifest)$/)) {
       // Meh, resources
-      continue;
+      return;
     }
-    const isGitHub = trimmed.match(/^https?:\/\/(?:www\.)?github\.com/);
```

Comment:
I removed the Github and learn.graphile checks because they don't appear to actually be doing anything? Maybe initially you wanted to limit what external urls were allows? or wanted to do special checks for github/learn urls based on path or something like we are for postgres? But it wasn't being used. So I removed them

Reply:
Specifically they were whitelisted I think 👍